str/str_iterative_hipstr_runner.py

```diff
@@ -0,0 +1,142 @@
+BCFTOOLS_IMAGE = config['images']['bcftools']
+
+
+def get_cloudfuse_paths(dataset, external_wgs_ids):
+    """Retrieves cloud fuse paths and outputs as a comma-separated string for crams associated with the external-wgs-ids"""
+    # Extract CPG IDs from external sample IDs
+    external_id_to_cpg_id: dict[str, str] = SampleApi().get_sample_id_map_by_external(
+        dataset, list(external_wgs_ids)
+    )
+    if dataset == 'tob-wgs':
+        analysis_query_model = AnalysisQueryModel(
+            sample_ids=list(external_id_to_cpg_id.values()),
+            projects=[dataset],
+            type=AnalysisType('cram'),
+            status=AnalysisStatus('completed'),
+            meta={'sequence_type': 'genome', 'source': 'nagim'},
+        )
+    else: 
+        analysis_query_model = AnalysisQueryModel(
+            sample_ids=list(external_id_to_cpg_id.values()),
+            projects=[dataset],
+            type=AnalysisType('cram'),
+            status=AnalysisStatus('completed'),
+            meta={},
+        )
+    crams_path = AnalysisApi().query_analyses(analysis_query_model)
+    cpg_sids_with_crams = set(sid for sids in crams_path for sid in sids['sample_ids'])
+    cpg_id_to_external_id = {
+        cpg_id: external_wgs_id
+        for external_wgs_id, cpg_id in external_id_to_cpg_id.items()
+    }
+    cpg_sids_without_crams = set(cpg_id_to_external_id.keys()) - cpg_sids_with_crams
+    if cpg_sids_without_crams:
+        external_wgs_sids_without_crams = ', '.join(
+            cpg_id_to_external_id[sid] for sid in cpg_sids_without_crams
+        )
+        logging.warning(
+            f'There were some samples without CRAMs: {external_wgs_sids_without_crams}'
+        )
+    # Create string containing paths based on /cramfuse
+    cramfuse_path = []
+    for cram_obj in crams_path:
+        suffix = cram_obj['output'].removeprefix('gs://').split('/', maxsplit=1)[1]
+        cramfuse_path.append(f'/cramfuse/{suffix}')
+    cramfuse_path = ','.join(cramfuse_path) # string format for input into hipstr 
+    return cramfuse_path
+
+# inputs:
+@click.option('--variant-catalog', help='Full path to HipSTR Variants catalog')
+@click.option('--dataset', help='dataset eg tob-wgs')
+@click.argument('external-wgs-ids', nargs=-1)
+@click.option('--output-file-name', help='Output file name without file extension')
+@click.command()
+def main(
+    variant_catalog, dataset, external_wgs_ids, output_file_name
+):  # pylint: disable=missing-function-docstring
+    b = get_batch()
+    # Create HipSTR job
+    hipstr_job = b.new_job(name=f'HipSTR running')
+    hipstr_job.image(HIPSTR_IMAGE)
+    hipstr_job.storage('375G')
+    hipstr_job.cpu(16)
+
+    hipstr_job.declare_resource_group(
+        hipstr_output={
+            'vcf.gz': '{root}.vcf.gz',
+            'viz.gz': '{root}.viz.gz',
+            'log.txt': '{root}.log.txt',
+        }
+    )
+    hipstr_job.cloudfuse(f'cpg-{dataset}-main', '/cramfuse')
```

Comment:
@lgruen, is there any issue with binding the whole main disk on? Does it do anything if you don't access any files, or does it cause a listing if you only access specific files?

If so, you could simplify using `os.path.commonprefix`

Reply:
Mounting the whole bucket should be okay, i.e. it shouldn't cause a full listing. The semantics with GCSFuse can be a bit tricky, see e.g. https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/docs/semantics.md#implicit-directories.